cmd/gateway/commands.go

```diff
@@ -168,6 +181,13 @@ func createStaticModeCommand() *cobra.Command {
 			"equal, it will choose the resource that appears first in alphabetical order by {namespace}/{name}.",
 	)
 
+	cmd.Flags().StringVar(
+		&configCRDName,
+		"nginx-gateway-config-name",
```

Comment:
Is that redundant though? It's assumed that a resource in k8s is Config. That's what everything is.

Reply:
not sure. but we end up using the `config` suffix  everywhere -- from cli arg to variables in the code. This makes me think why not add that suffix to resource kind.

For other resources, we don't use suffixes like that. For example, for HTTPRoute or Gateway resources.

More over, we also have a `-gateway` cli arg, the name of which I think describes it well.

I wonder if we also want to drop the name suffix, for consistency, because `-gateway` doesn't include nsname suffix or something like that.

Reply:
I think the cli flag could just be `config` or `config-name`. I think that's descriptive and easy for users to understand. We could even add a short name `-c`. 

I'm fine with using either `NginxGateway` or `NginxGatewayConfig` for the resource name. Envoy handled this by adding `config` to their api name `config.gateway.envoyproxy.io`, which is also an option.

Reply:
@pleshakov With the field being named `config`, are you okay with keeping the CRD as NginxGateway?